src/tovnode.ts

```diff
@@ -2,13 +2,25 @@ import { addNS } from "./h";
 import { vnode, VNode } from "./vnode";
 import { htmlDomApi, DOMAPI } from "./htmldomapi";
 
+/**
+ * Transforms the given attribute name into a valid dataset property key
+ * according to https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset#name_conversion
+ *
+ * @param attributeName data- attribute name, must start with data-
+ */
+function datasetKey(attributeName: string) {
+  return attributeName
+    .slice(5)
+    .replace(/-([a-z])/g, (_, ch) => ch.toUpperCase());
```

Comment:
I think this doesn't carry out the first step of the [linked conversion](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset#name_conversion). But I guess that's not very important.